File: TOMIMSV4/DAL/Transacciones/OrdenCompra/Embarcador/clsLnTrans_oc_embarcador.vb
Imports System.Data.SqlClient
Imports System.Reflection

Public Class clsLnTrans_oc_embarcador

    Public Shared Sub Cargar(ByRef oBeTrans_oc_embarcador As clsBeTrans_oc_embarcador, ByRef dr As DataRow)
        Try
            With oBeTrans_oc_embarcador
                .IdEmbarcador = IIf(IsDBNull(dr.Item("IdEmbarcador")), 0, dr.Item("IdEmbarcador"))
                .Codigo = IIf(IsDBNull(dr.Item("Codigo")), "", dr.Item("Codigo"))
                .Nombre = IIf(IsDBNull(dr.Item("Nombre")), "", dr.Item("Nombre"))
            End With
        Catch ex As Exception
            Throw New Exception(String.Format("{0} {1}", MethodBase.GetCurrentMethod.Name(), ex.Message))
        End Try
    End Sub

    Public Shared Function Insertar(ByRef oBeTrans_oc_embarcador As clsBeTrans_oc_embarcador, Optional ByVal pConection as SqlConnection = Nothing, Optional Byval pTransaction as SqlTransaction = Nothing) As Integer

        Dim lConnection As New SqlConnection(connectionString:=Configuration.ConfigurationManager.AppSettings("CST"))
        Dim lTransaction As SqlTransaction = Nothing

        Try

            Ins.Init("trans_oc_embarcador")
            Ins.Add("idembarcador", "@idembarcador", DataType.Parametro)
            Ins.Add("codigo", "@codigo", DataType.Parametro)
            Ins.Add("nombre", "@nombre", DataType.Parametro)

            Dim sp As String = Ins.SQL()
            Dim cmd As New SqlCommand With {.CommandType = CommandType.Text}

            Dim Es_Transaccion_Remota As Boolean = (Not pConection Is Nothing AndAlso Not pTransaction Is Nothing)

            If Es_Transaccion_Remota Then
                cmd = New SqlCommand(sp, pConection, pTransaction)
            Else
                lConnection.Open() : lTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUnCommitted)
                cmd = New SqlCommand(sp, lConnection, lTransaction)
            End If

            cmd.Parameters.Add(New SqlParameter("@IDEMBARCADOR", oBeTrans_oc_embarcador.IdEmbarcador))
            cmd.Parameters.Add(New SqlParameter("@CODIGO", oBeTrans_oc_embarcador.Codigo))
            cmd.Parameters.Add(New SqlParameter("@NOMBRE", oBeTrans_oc_embarcador.Nombre))

            Dim rowsAffected As Integer = cmd.ExecuteNonQuery()

            cmd.Dispose

            If Not Es_Transaccion_Remota Then lTransaction.Commit()

            Return rowsAffected

        Catch ex As Exception
            If Not lTransaction Is Nothing Then lTransaction.Rollback()
            Throw New Exception(String.Format("{0} {1}", MethodBase.GetCurrentMethod.Name(), ex.Message))
        Finally
            If lConnection.State = ConnectionState.Open Then lConnection.Close
            If Not lConnection is Nothing Then lConnection.Dispose()
            If Not lTransaction is Nothing Then lTransaction.Dispose()
        End Try

    End Function

    Public Shared Function Actualizar(ByRef oBeTrans_oc_embarcador As clsBeTrans_oc_embarcador, Optional ByVal pConection as SqlConnection = Nothing, Optional Byval pTransaction as SqlTransaction = Nothing) As Integer

        Dim lConnection As New SqlConnection(connectionString:=Configuration.ConfigurationManager.AppSettings("CST"))
        Dim lTransaction As SqlTransaction = Nothing

        Try

            Upd.Init("trans_oc_embarcador")
            Upd.Add("idembarcador", "@idembarcador", DataType.Parametro)
            Upd.Add("codigo", "@codigo", DataType.Parametro)
            Upd.Add("nombre", "@nombre", DataType.Parametro)
            Upd.Where("IdEmbarcador = @IdEmbarcador")

            Dim sp As String = Upd.SQL()

            Dim Es_Transaccion_Remota As Boolean = (Not pConection Is Nothing AndAlso Not pTransaction Is Nothing)

            Dim cmd As New SqlCommand With {.CommandType = CommandType.Text}

            If Es_Transaccion_Remota then
                cmd = New SqlCommand(sp, pConection, pTransaction)
            Else
                lConnection.Open() : lTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUnCommitted)
                cmd = New SqlCommand(sp, lConnection, lTransaction)
            End If

            cmd.Parameters.Add(New SqlParameter("@IDEMBARCADOR", oBeTrans_oc_embarcador.IdEmbarcador))
            cmd.Parameters.Add(New SqlParameter("@CODIGO", oBeTrans_oc_embarcador.Codigo))
            cmd.Parameters.Add(New SqlParameter("@NOMBRE", oBeTrans_oc_embarcador.Nombre))

            Dim rowsAffected As Integer = cmd.ExecuteNonQuery()

            cmd.Dispose

            If Not Es_Transaccion_Remota Then lTransaction.Commit()

            Return rowsAffected

        Catch ex As Exception
            If Not lTransaction Is Nothing Then lTransaction.Rollback()
            Throw New Exception(String.Format("{0} {1}", MethodBase.GetCurrentMethod.Name(), ex.Message))
        Finally
            If lConnection.State = ConnectionState.Open Then lConnection.Close
            If Not lConnection is Nothing Then lConnection.Dispose()
            If Not lTransaction is Nothing Then lTransaction.Dispose()
        End Try

    End Function


    Public Shared Function Eliminar(ByRef oBeTrans_oc_embarcador As clsBeTrans_oc_embarcador, Optional ByVal pConection as SqlConnection = Nothing, Optional Byval pTransaction as SqlTransaction = Nothing) As Integer

        Dim lConnection As New SqlConnection(connectionString:=Configuration.ConfigurationManager.AppSettings("CST"))
        Dim lTransaction As SqlTransaction = Nothing

        Try

            Const sp As String = " Delete from Trans_oc_embarcador" & _
             "  Where(IdEmbarcador = @IdEmbarcador)"

            Dim cmd As New SqlCommand With {.CommandType = CommandType.Text}

            Dim Es_Transaccion_Remota As Boolean = (Not pConection Is Nothing AndAlso Not pTransaction Is Nothing)

            If Es_Transaccion_Remota then
                cmd = New SqlCommand(sp, pConection, pTransaction)
            Else
                lConnection.Open() : lTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUnCommitted)
                cmd = New SqlCommand(sp, lConnection, lTransaction)
            End If

            cmd.Parameters.Add(New SqlParameter("@IDEMBARCADOR", oBeTrans_oc_embarcador.IdEmbarcador))

            Dim rowsAffected As Integer = cmd.ExecuteNonQuery()

            cmd.Dispose

            If Not Es_Transaccion_Remota Then lTransaction.Commit()

            Return rowsAffected

        Catch ex As Exception
            If Not lTransaction Is Nothing Then lTransaction.Rollback()
            Throw New Exception(String.Format("{0} {1}", MethodBase.GetCurrentMethod.Name(), ex.Message))
        Finally
            If lConnection.State = ConnectionState.Open Then lConnection.Close
            If Not lConnection is Nothing Then lConnection.Dispose()
            If Not lTransaction is Nothing Then lTransaction.Dispose()
        End Try

    End Function

    Public Shared Function Listar() As DataTable

        Dim lConnection As New SqlConnection(connectionString:=Configuration.ConfigurationManager.AppSettings("CST"))
        Dim lTransaction As SqlTransaction = Nothing

        Try

            Const sp As String = "SELECT * FROM Trans_oc_embarcador"
            lConnection.Open() : lTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUnCommitted)
            Dim cmd As New SqlCommand(sp, lConnection, lTransaction) With {.CommandType = CommandType.Text}
            Dim dad As New SqlDataAdapter(cmd)
            Dim dt As New DataTable
            dad.Fill(dt)

            lTransaction.Commit()

            Return dt

        Catch ex As Exception
            If Not lTransaction Is Nothing Then lTransaction.Rollback()
            Throw New Exception(String.Format("{0} {1}", MethodBase.GetCurrentMethod.Name(), ex.Message))
        Finally
            If lConnection.State = ConnectionState.Open Then lConnection.Close
            If Not lConnection is Nothing Then lConnection.Dispose()
            If Not lTransaction is Nothing Then lTransaction.Dispose()
        End Try

    End Function

    Public Shared Function Get_All() As List(Of clsBeTrans_oc_embarcador)

        Dim lReturnList As New List(Of clsBeTrans_oc_embarcador)

        Try

            Const sp As String = "SELECT * FROM Trans_oc_embarcador"

            Using lConnection As New SqlConnection(connectionString:=Configuration.ConfigurationManager.AppSettings("CST"))

                lConnection.Open()

                Using lTransaction As SqlTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUnCommitted)

                    Using lDTA As New SqlDataAdapter(sp, lConnection)

                        lDTA.SelectCommand.CommandType = CommandType.Text
                        lDTA.SelectCommand.Transaction = lTransaction
                        Dim lDataTable As New DataTable
                        lDTA.Fill(lDataTable)

                        Dim vBeTrans_oc_embarcador As New clsBeTrans_oc_embarcador

                        For Each dr As DataRow In lDataTable.Rows
                            vBeTrans_oc_embarcador = New clsBeTrans_oc_embarcador()
                            Cargar(vBeTrans_oc_embarcador, dr)
                            lReturnList.Add(vBeTrans_oc_embarcador)
                        Next

                    End Using

                    lTransaction.Commit()

                End Using

                lConnection.Close()

            End Using

            Return lReturnList

        Catch ex As Exception
            Throw New Exception(String.Format("{0} {1}", MethodBase.GetCurrentMethod.Name(), ex.Message))
        End Try

    End Function

    Public Shared Sub GetSingle(ByRef pBeTrans_oc_embarcador As clsBeTrans_oc_embarcador)

        Try

            Const sp As String = "SELECT * FROM Trans_oc_embarcador" & _
            " Where(IdEmbarcador = @IdEmbarcador)"


            Using lConnection As New SqlConnection(connectionString:=Configuration.ConfigurationManager.AppSettings("CST"))

                lConnection.Open()

                Using lTransaction As SqlTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUnCommitted)

                    Using lDTA As New SqlDataAdapter(sp, lConnection)

                        lDTA.SelectCommand.CommandType = CommandType.Text
                        lDTA.SelectCommand.Transaction = lTransaction
                        Dim lDataTable As New DataTable
                        lDTA.Fill(lDataTable)

                        Dim vBeTrans_oc_embarcador As New clsBeTrans_oc_embarcador

                        If lDataTable IsNot Nothing AndAlso lDataTable.Rows.Count > 0 Then
                            Cargar(vBeTrans_oc_embarcador, lDataTable.Rows(0))
                        End If

                    End Using

                    lTransaction.Commit()

                End Using

                lConnection.Close()

            End Using

        Catch ex As Exception
            Throw New Exception(String.Format("{0} {1}", MethodBase.GetCurrentMethod.Name(), ex.Message))
        End Try

    End Sub

    Public Shared Function MaxID() as Integer

        Try

            Dim lMax As Integer = 0

            Const sp As String = "SELECT ISNULL(Max(IdEmbarcador),0) FROM Trans_oc_embarcador"

            Using lConnection As New SqlConnection(connectionString:=Configuration.ConfigurationManager.AppSettings("CST"))

                lConnection.Open()

                Using lTransaction As SqlTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUnCommitted)

                    Using lCommand As New SqlCommand(sp, lConnection, lTransaction) With {.CommandType = CommandType.Text}

                        Dim lReturnValue As Object = lCommand.ExecuteScalar()
                        If lReturnValue IsNot DBNull.Value AndAlso lReturnValue IsNot Nothing Then
                            lMax = CInt(lReturnValue)
                        End If

                    End Using

                    lTransaction.Commit()

                End Using

                lConnection.Close()

            End Using

            Return lMax

        Catch ex As Exception
            Throw New Exception(String.Format("{0} {1}", MethodBase.GetCurrentMethod.Name(), ex.Message))
        End Try

    End Function

    Public Shared Function Get_Single_By_Nombre(ByVal pNombre As String) As clsBeTrans_oc_embarcador

        Get_Single_By_Nombre = Nothing

        Try

            Const sp As String = "SELECT * FROM Trans_oc_embarcador 
								  Where(Nombre = @Nombre) "


            Using lConnection As New SqlConnection(connectionString:=Configuration.ConfigurationManager.AppSettings("CST"))

                lConnection.Open()

                Using lTransaction As SqlTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUncommitted)

                    Using lDTA As New SqlDataAdapter(sp, lConnection)

                        lDTA.SelectCommand.CommandType = CommandType.Text
                        lDTA.SelectCommand.Transaction = lTransaction
                        lDTA.SelectCommand.Parameters.AddWithValue("@Nombre", pNombre.Trim)

                        Dim lDataTable As New DataTable
                        lDTA.Fill(lDataTable)

                        Dim vBeTrans_oc_embarcador As New clsBeTrans_oc_embarcador

                        If lDataTable IsNot Nothing AndAlso lDataTable.Rows.Count > 0 Then
                            Cargar(vBeTrans_oc_embarcador, lDataTable.Rows(0))
                            Get_Single_By_Nombre = vBeTrans_oc_embarcador
                        End If

                    End Using

                    lTransaction.Commit()

                End Using

                lConnection.Close()

            End Using

        Catch ex As Exception
            Throw ex
        End Try

    End Function

    Public Shared Function Get_Single_By_Nombre(ByVal pNombre As String,
                                                ByVal lConnection As SqlConnection,
                                                ByVal lTransaction As SqlTransaction) As clsBeTrans_oc_embarcador

        Get_Single_By_Nombre = Nothing

        Try

            Const sp As String = "SELECT * FROM Trans_oc_embarcador 
								  Where(Nombre = @Nombre) "


            Using lDTA As New SqlDataAdapter(sp, lConnection)

                lDTA.SelectCommand.CommandType = CommandType.Text
                lDTA.SelectCommand.Transaction = lTransaction
                lDTA.SelectCommand.Parameters.AddWithValue("@Nombre", pNombre.Trim)

                Dim lDataTable As New DataTable
                lDTA.Fill(lDataTable)

                Dim vBeTrans_oc_embarcador As New clsBeTrans_oc_embarcador

                If lDataTable IsNot Nothing AndAlso lDataTable.Rows.Count > 0 Then
                    Cargar(vBeTrans_oc_embarcador, lDataTable.Rows(0))
                    Get_Single_By_Nombre = vBeTrans_oc_embarcador
                End If

            End Using

        Catch ex As Exception
            Throw ex
        End Try

    End Function

    Public Shared Function MaxID(ByVal lConnection As SqlConnection, ByVal lTransaction As SqlTransaction) As Integer

        Try

            Dim lMax As Integer = 0

            Const sp As String = "SELECT ISNULL(Max(IdEmbarcador),0) FROM Trans_oc_embarcador "

            Using lCommand As New SqlCommand(sp, lConnection, lTransaction) With {.CommandType = CommandType.Text}

                Dim lReturnValue As Object = lCommand.ExecuteScalar()
                If lReturnValue IsNot DBNull.Value AndAlso lReturnValue IsNot Nothing Then
                    lMax = CInt(lReturnValue)
                End If

            End Using

            Return lMax

        Catch ex As Exception
            Throw ex
        End Try

    End Function

    Public Shared Function Get_Single_By_IdEmbarcador(ByVal pIdEmbarcador As Integer) As clsBeTrans_oc_embarcador

        Get_Single_By_IdEmbarcador = Nothing

        Try

            Const sp As String = "SELECT * FROM Trans_oc_embarcador" &
            " Where(IdEmbarcador = @IdEmbarcador)"


            Using lConnection As New SqlConnection(connectionString:=Configuration.ConfigurationManager.AppSettings("CST"))

                lConnection.Open()

                Using lTransaction As SqlTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUncommitted)

                    Using lDTA As New SqlDataAdapter(sp, lConnection)

                        lDTA.SelectCommand.CommandType = CommandType.Text
                        lDTA.SelectCommand.Transaction = lTransaction
                        lDTA.SelectCommand.Parameters.AddWithValue("@IdEmbarcador", pIdEmbarcador)

                        Dim lDataTable As New DataTable
                        lDTA.Fill(lDataTable)

                        Dim vBeTrans_oc_embarcador As New clsBeTrans_oc_embarcador

                        If lDataTable IsNot Nothing AndAlso lDataTable.Rows.Count > 0 Then
                            Cargar(vBeTrans_oc_embarcador, lDataTable.Rows(0))
                        End If

                    End Using

                    lTransaction.Commit()

                End Using

                lConnection.Close()

            End Using

        Catch ex As Exception
            Throw ex
        End Try

    End Function

    Public Shared Function Get_Single_By_IdEmbarcador(ByVal pIdEmbarcador As Integer,
                                                      ByVal lConnection As SqlConnection,
                                                      ByVal lTransaction As SqlTransaction) As clsBeTrans_oc_embarcador

        Get_Single_By_IdEmbarcador = Nothing

        Try

            Const sp As String = "SELECT * FROM Trans_oc_embarcador 
								  Where(IdEmbarcador = @IdEmbarcador) "

            Using lDTA As New SqlDataAdapter(sp, lConnection)

                lDTA.SelectCommand.CommandType = CommandType.Text
                lDTA.SelectCommand.Transaction = lTransaction
                lDTA.SelectCommand.Parameters.AddWithValue("@IdEmbarcador", pIdEmbarcador)

                Dim lDataTable As New DataTable
                lDTA.Fill(lDataTable)

                Dim vBeTrans_oc_embarcador As New clsBeTrans_oc_embarcador

                If lDataTable IsNot Nothing AndAlso lDataTable.Rows.Count > 0 Then
                    Cargar(vBeTrans_oc_embarcador, lDataTable.Rows(0))
                    Get_Single_By_IdEmbarcador = vBeTrans_oc_embarcador
                End If

            End Using

        Catch ex As Exception
            Throw ex
        End Try

    End Function

End Class

File: WMS.DALCore/Trans_oc/clsLnTrans_oc_embarcador.cs
namespace WMS.DALCore.Trans_oc
{
    using System;
    using System.Data;
    using System.Collections.Generic;
    using WMS.EntityCore.Trans_oc;
    using Microsoft.Data.SqlClient;
    using Microsoft.Extensions.Configuration;

    public class clsLnTrans_oc_embarcador
    {
        private static clsInsert Ins = new clsInsert();
        private static clsUpdate Upd = new clsUpdate();
        public static void Cargar(ref clsBeTrans_oc_embarcador oBeTrans_oc_embarcador, DataRow dr)
        {
            try
            {
                oBeTrans_oc_embarcador.IdEmbarcador = (dr["IdEmbarcador"] == DBNull.Value) ? 0 : Convert.ToInt32(dr["IdEmbarcador"]);
                oBeTrans_oc_embarcador.Codigo = (dr["Codigo"] == DBNull.Value) ? "" : Convert.ToString(dr["Codigo"]) ?? "";
                oBeTrans_oc_embarcador.Nombre = (dr["Nombre"] == DBNull.Value) ? "" : Convert.ToString(dr["Nombre"]) ?? "";
            }
            catch (Exception)
            {
                throw;
            }
        }

        public static int Insertar(ref clsBeTrans_oc_embarcador oBeTrans_oc_embarcador, SqlConnection pConection, SqlTransaction pTransaction)
        {
            SqlCommand cmd = new SqlCommand();

            try
            {
                Ins.Init("trans_oc_embarcador");
                Ins.Add("idembarcador", "@idembarcador", "F");
                Ins.Add("codigo", "@codigo", "F");
                Ins.Add("nombre", "@nombre", "F");

                string sp = Ins.SQL();
                cmd = new SqlCommand(sp, pConection, pTransaction) { CommandType = CommandType.Text };

                cmd.Parameters.Add(new SqlParameter("@IDEMBARCADOR", oBeTrans_oc_embarcador.IdEmbarcador));
                cmd.Parameters.Add(new SqlParameter("@CODIGO", oBeTrans_oc_embarcador.Codigo));
                cmd.Parameters.Add(new SqlParameter("@NOMBRE", oBeTrans_oc_embarcador.Nombre));

                int rowsAffected = cmd.ExecuteNonQuery();
                return rowsAffected;
            }
            catch (Exception)
            {
                throw;
            }
            finally
            {
                cmd?.Dispose();
            }
        }

        public static int Actualizar(ref clsBeTrans_oc_embarcador oBeTrans_oc_embarcador, SqlConnection pConection, SqlTransaction pTransaction)
        {
            SqlCommand cmd = new SqlCommand();

            try
            {
                Upd.Init("trans_oc_embarcador");
                Upd.Add("idembarcador", "@idembarcador", "F");
                Upd.Add("codigo", "@codigo", "F");
                Upd.Add("nombre", "@nombre", "F");
                Upd.Where("IdEmbarcador = @IdEmbarcador");

                string sp = Upd.SQL();
                cmd = new SqlCommand(sp, pConection, pTransaction) { CommandType = CommandType.Text };

                cmd.Parameters.Add(new SqlParameter("@IDEMBARCADOR", oBeTrans_oc_embarcador.IdEmbarcador));
                cmd.Parameters.Add(new SqlParameter("@CODIGO", oBeTrans_oc_embarcador.Codigo));
                cmd.Parameters.Add(new SqlParameter("@NOMBRE", oBeTrans_oc_embarcador.Nombre));

                int rowsAffected = cmd.ExecuteNonQuery();
                return rowsAffected;
            }
            catch (Exception)
            {
                throw;
            }
            finally
            {
                cmd?.Dispose();
            }
        }

        public static int Eliminar(ref clsBeTrans_oc_embarcador oBeTrans_oc_embarcador, SqlConnection pConection, SqlTransaction pTransaction)
        {
            SqlCommand cmd = new SqlCommand();

            try
            {
                const string sp = " Delete from Trans_oc_embarcador Where(IdEmbarcador = @IdEmbarcador)";
                cmd = new SqlCommand(sp, pConection, pTransaction) { CommandType = CommandType.Text };

                cmd.Parameters.Add(new SqlParameter("@IDEMBARCADOR", oBeTrans_oc_embarcador.IdEmbarcador));

                int rowsAffected = cmd.ExecuteNonQuery();
                return rowsAffected;
            }
            catch (Exception)
            {
                throw;
            }
            finally
            {
                cmd?.Dispose();
            }
        }    

        public static List<clsBeTrans_oc_embarcador> Get_All(IConfiguration config)
        {
            List<clsBeTrans_oc_embarcador> lReturnList = new List<clsBeTrans_oc_embarcador>();

            try
            {
                const string sp = "SELECT * FROM Trans_oc_embarcador";

                using (SqlConnection lConnection = new SqlConnection(config.GetConnectionString("CST")))
                {
                    lConnection.Open();

                    using (SqlTransaction lTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUncommitted))
                    {
                        using (SqlDataAdapter lDTA = new SqlDataAdapter(sp, lConnection))
                        {
                            lDTA.SelectCommand.CommandType = CommandType.Text;
                            lDTA.SelectCommand.Transaction = lTransaction;
                            DataTable lDataTable = new DataTable();
                            lDTA.Fill(lDataTable);

                            clsBeTrans_oc_embarcador vBeTrans_oc_embarcador;

                            foreach (DataRow dr in lDataTable.Rows)
                            {
                                vBeTrans_oc_embarcador = new clsBeTrans_oc_embarcador();
                                Cargar(ref vBeTrans_oc_embarcador, dr);
                                lReturnList.Add(vBeTrans_oc_embarcador);
                            }
                        }

                        lTransaction.Commit();
                    }

                    lConnection.Close();
                }

                return lReturnList;
            }
            catch (Exception)
            {
                throw;
            }
        }

        public static void GetSingle(IConfiguration config, ref clsBeTrans_oc_embarcador pBeTrans_oc_embarcador)
        {
            try
            {
                const string sp = "SELECT * FROM Trans_oc_embarcador Where(IdEmbarcador = @IdEmbarcador)";

                using (SqlConnection lConnection = new SqlConnection(config.GetConnectionString("CST")))
                {
                    lConnection.Open();

                    using (SqlTransaction lTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUncommitted))
                    {
                        using (SqlDataAdapter lDTA = new SqlDataAdapter(sp, lConnection))
                        {
                            lDTA.SelectCommand.CommandType = CommandType.Text;
                            lDTA.SelectCommand.Transaction = lTransaction;
                            DataTable lDataTable = new DataTable();
                            lDTA.Fill(lDataTable);

                            clsBeTrans_oc_embarcador vBeTrans_oc_embarcador = new clsBeTrans_oc_embarcador();

                            if (lDataTable != null && lDataTable.Rows.Count > 0)
                            {
                                Cargar(ref vBeTrans_oc_embarcador, lDataTable.Rows[0]);
                            }
                        }

                        lTransaction.Commit();
                    }

                    lConnection.Close();
                }
            }
            catch (Exception)
            {
                throw;
            }
        }

        public static int MaxID(IConfiguration config)
        {
            try
            {
                int lMax = 0;
                const string sp = "SELECT ISNULL(Max(IdEmbarcador),0) FROM Trans_oc_embarcador";

                using (SqlConnection lConnection = new SqlConnection(config.GetConnectionString("CST")))
                {
                    lConnection.Open();

                    using (SqlTransaction lTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUncommitted))
                    {
                        using (SqlCommand lCommand = new SqlCommand(sp, lConnection, lTransaction) { CommandType = CommandType.Text })
                        {
                            object lReturnValue = lCommand.ExecuteScalar();
                            if (lReturnValue != DBNull.Value && lReturnValue != null)
                            {
                                lMax = Convert.ToInt32(lReturnValue);
                            }
                        }

                        lTransaction.Commit();
                    }

                    lConnection.Close();
                }

                return lMax;
            }
            catch (Exception)
            {
                throw;
            }
        }

        public static clsBeTrans_oc_embarcador? Get_Single_By_Nombre(IConfiguration config, string pNombre)
        {
            clsBeTrans_oc_embarcador? result = null;

            try
            {
                const string sp = "SELECT * FROM Trans_oc_embarcador Where(Nombre = @Nombre) ";

                using (SqlConnection lConnection = new SqlConnection(config.GetConnectionString("CST")))
                {
                    lConnection.Open();

                    using (SqlTransaction lTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUncommitted))
                    {
                        using (SqlDataAdapter lDTA = new SqlDataAdapter(sp, lConnection))
                        {
                            lDTA.SelectCommand.CommandType = CommandType.Text;
                            lDTA.SelectCommand.Transaction = lTransaction;
                            lDTA.SelectCommand.Parameters.AddWithValue("@Nombre", pNombre.Trim());

                            DataTable lDataTable = new DataTable();
                            lDTA.Fill(lDataTable);

                            clsBeTrans_oc_embarcador vBeTrans_oc_embarcador = new clsBeTrans_oc_embarcador();

                            if (lDataTable != null && lDataTable.Rows.Count > 0)
                            {
                                Cargar(ref vBeTrans_oc_embarcador, lDataTable.Rows[0]);
                                result = vBeTrans_oc_embarcador;
                            }
                        }

                        lTransaction.Commit();
                    }

                    lConnection.Close();
                }
            }
            catch (Exception)
            {
                throw;
            }

            return result;
        }

        public static clsBeTrans_oc_embarcador? Get_Single_By_Nombre(string pNombre, SqlConnection lConnection, SqlTransaction lTransaction)
        {
            clsBeTrans_oc_embarcador? result = null;

            try
            {
                const string sp = "SELECT * FROM Trans_oc_embarcador Where(Nombre = @Nombre) ";

                using (SqlDataAdapter lDTA = new SqlDataAdapter(sp, lConnection))
                {
                    lDTA.SelectCommand.CommandType = CommandType.Text;
                    lDTA.SelectCommand.Transaction = lTransaction;
                    lDTA.SelectCommand.Parameters.AddWithValue("@Nombre", pNombre.Trim());

                    DataTable lDataTable = new DataTable();
                    lDTA.Fill(lDataTable);

                    clsBeTrans_oc_embarcador vBeTrans_oc_embarcador = new clsBeTrans_oc_embarcador();

                    if (lDataTable != null && lDataTable.Rows.Count > 0)
                    {
                        Cargar(ref vBeTrans_oc_embarcador, lDataTable.Rows[0]);
                        result = vBeTrans_oc_embarcador;
                    }
                }
            }
            catch (Exception)
            {
                throw;
            }

            return result;
        }

        public static int MaxID(SqlConnection lConnection, SqlTransaction lTransaction)
        {
            try
            {
                int lMax = 0;
                const string sp = "SELECT ISNULL(Max(IdEmbarcador),0) FROM Trans_oc_embarcador ";

                using (SqlCommand lCommand = new SqlCommand(sp, lConnection, lTransaction) { CommandType = CommandType.Text })
                {
                    object lReturnValue = lCommand.ExecuteScalar();
                    if (lReturnValue != DBNull.Value && lReturnValue != null)
                    {
                        lMax = Convert.ToInt32(lReturnValue);
                    }
                }

                return lMax;
            }
            catch (Exception)
            {
                throw;
            }
        }

        public static clsBeTrans_oc_embarcador? Get_Single_By_IdEmbarcador(IConfiguration config, int pIdEmbarcador)
        {
            clsBeTrans_oc_embarcador? result = null;

            try
            {
                const string sp = "SELECT * FROM Trans_oc_embarcador Where(IdEmbarcador = @IdEmbarcador)";

                using (SqlConnection lConnection = new SqlConnection(config.GetConnectionString("CST")))
                {
                    lConnection.Open();

                    using (SqlTransaction lTransaction = lConnection.BeginTransaction(IsolationLevel.ReadUncommitted))
                    {
                        using (SqlDataAdapter lDTA = new SqlDataAdapter(sp, lConnection))
                        {
                            lDTA.SelectCommand.CommandType = CommandType.Text;
                            lDTA.SelectCommand.Transaction = lTransaction;
                            lDTA.SelectCommand.Parameters.AddWithValue("@IdEmbarcador", pIdEmbarcador);

                            DataTable lDataTable = new DataTable();
                            lDTA.Fill(lDataTable);

                            clsBeTrans_oc_embarcador vBeTrans_oc_embarcador = new clsBeTrans_oc_embarcador();

                            if (lDataTable != null && lDataTable.Rows.Count > 0)
                            {
                                Cargar(ref vBeTrans_oc_embarcador, lDataTable.Rows[0]);
                                result = vBeTrans_oc_embarcador;
                            }
                        }

                        lTransaction.Commit();
                    }

                    lConnection.Close();
                }
            }
            catch (Exception)
            {
                throw;
            }

            return result;
        }

        public static clsBeTrans_oc_embarcador? Get_Single_By_IdEmbarcador(int pIdEmbarcador, SqlConnection lConnection, SqlTransaction lTransaction)
        {
            clsBeTrans_oc_embarcador? result = null;

            try
            {
                const string sp = "SELECT * FROM Trans_oc_embarcador Where(IdEmbarcador = @IdEmbarcador) ";

                using (SqlDataAdapter lDTA = new SqlDataAdapter(sp, lConnection))
                {
                    lDTA.SelectCommand.CommandType = CommandType.Text;
                    lDTA.SelectCommand.Transaction = lTransaction;
                    lDTA.SelectCommand.Parameters.AddWithValue("@IdEmbarcador", pIdEmbarcador);

                    DataTable lDataTable = new DataTable();
                    lDTA.Fill(lDataTable);

                    clsBeTrans_oc_embarcador vBeTrans_oc_embarcador = new clsBeTrans_oc_embarcador();

                    if (lDataTable != null && lDataTable.Rows.Count > 0)
                    {
                        Cargar(ref vBeTrans_oc_embarcador, lDataTable.Rows[0]);
                        result = vBeTrans_oc_embarcador;
                    }
                }
            }
            catch (Exception)
            {
                throw;
            }

            return result;
        }
    }
}
